spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/DefaultAmqpHeaderMapper.java

```diff
@@ -79,6 +95,57 @@
 		STANDARD_HEADER_NAMES.add(JsonHeaders.KEY_TYPE_ID);
 		STANDARD_HEADER_NAMES.add(AmqpHeaders.SPRING_REPLY_CORRELATION);
 		STANDARD_HEADER_NAMES.add(AmqpHeaders.SPRING_REPLY_TO_STACK);
+
+		final AtomicReference<String> tag = new AtomicReference<String>();
+		final AtomicReference<String> queue = new AtomicReference<String>();
+		final AtomicReference<Method> tagMethod = new AtomicReference<Method>();
+		final AtomicReference<Method> queueMethod = new AtomicReference<Method>();
+		try {
+			ReflectionUtils.doWithFields(AmqpHeaders.class, new FieldCallback() {
+
+				@Override
+				public void doWith(Field field) throws IllegalArgumentException, IllegalAccessException {
+					if ("CONSUMER_TAG".equals(field.getName())) {
+						tag.set((String) field.get(null));
+					}
+					else if ("CONSUMER_QUEUE".equals(field.getName())) {
+						queue.set((String) field.get(null));
+					}
+				}
+			},
+			new FieldFilter() {
+
+				@Override
+				public boolean matches(Field field) {
+					return field.getName().startsWith("CONSUMER_") && field.getType().equals(String.class);
+				}
+			});
+			ReflectionUtils.doWithMethods(MessageProperties.class, new MethodCallback() {
+
+				@Override
+				public void doWith(Method method) throws IllegalArgumentException, IllegalAccessException {
+					if (method.getName().equals("getConsumerTag")) {
+						tagMethod.set(method);
+					}
+					else if (method.getName().equals("getConsumerQueue")) {
+						queueMethod.set(method);
+					}
+
+				}
+			}, new MethodFilter() {
+
+				@Override
+				public boolean matches(Method method) {
+					return method.getName().startsWith("getConsumer") && method.getParameterCount() == 0;
+				}
+			});
+		}
+		catch (Exception e) {
+		}
+		CONSUMER_TAG_HEADER = tag.get();
+		CONSUMER_QUEUE_HEADER = queue.get();
+		CONSUMER_TAG_METHOD = tagMethod.get();
+		CONSUMER_QUEUE_METHOD = queueMethod.get();
```

Comment:
Well...
I think we can avoid this reflection hell just with single `AMQP_VERSION` property and make 1.4.2 as minimal for SI 4.1.2, since it is just a compile time issue. At runtime we won't call those new methods and use new `AmqpHeaders`, if we are with AMQP < 1.4.2.
WDYT ?